lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,313 @@
+	const manifest = JSON.parse(content);
+	const readdir = promisify(fs.readdir);
+
+	if (!manifest._version) {
+		log.verbose("manifest.json: _version is not defined. No supportedLocales are generated");
+		return;
+	}
+
+	const descriptorVersion = new Version(manifest._version);
+	if (descriptorVersion.compare(APP_DESCRIPTOR_V22) === -1) {
+		log.verbose("manifest.json: _version is lower than 1.21.0 so no supportedLocales can be generated");
+		return;
+	}
+
+	const project = resource.getProject();
+	const namespace = project.getNamespace();
+
+	// in case of usage by server middleware --> isNamespaced is false
+	const isNamespaced = namespace && resource.getPath().startsWith("/resources/" + namespace);
+
+	const bundles = getBundles(manifest);
+
+	if (bundles.length === 0) {
+		return;
+	}
+
+	await Promise.all(bundles.map(async (bundleConfig) => {
+		const {bundleUrl, bundleName, fallbackLocale} = bundleConfig;
+		let defaultBundleName = bundleUrl;
+
+		if (bundleName) {
+			defaultBundleName = bundleName.replace(/\./g, "/");
+			defaultBundleName = defaultBundleName.replace(namespace, "");
+		}
+
+		if (bundleUrl) {
+			defaultBundleName = bundleUrl.replace(".properties", "");
+
+			if (bundleUrl.startsWith("ui5://")) {
+				defaultBundleName = defaultBundleName.replace("ui5://", "").replace(namespace, "");
+			}
+		}
+
+		let pathToDefaultBundleName = "";
+
+		if (isNamespaced) {
+			pathToDefaultBundleName = path.join(`/resources/${namespace}`, defaultBundleName);
+		} else {
+			pathToDefaultBundleName = defaultBundleName.replace(namespace.replace(/\./g, "/"), "");
+		}
+		const generatedSupportedLocales = (await readdir(path.dirname(pathToDefaultBundleName)))
+			.filter((name) => name.endsWith(".properties") && name.startsWith(path.basename(defaultBundleName)))
+			.map((name) => {
+				return getLocale(name, defaultBundleName);
+			}).sort();
+
+		log.verbose("Generated supported locales: " + generatedSupportedLocales.join(", ").toString());
+
+		if (fallbackLocale && !isFallbackLocalePartOfSupportedLocales(fallbackLocale, generatedSupportedLocales)) {
+			log.error("manifest.json: Generated supported locales ('" + generatedSupportedLocales.join("', '") + "') " +
+				"not containing the defined fallback locale '" + fallbackLocale + "'. Either provide a " +
+				"properties file for defined fallbackLocale or configure another available fallbackLocale");
+			return;
+		} else if (!fallbackLocale && !isFallbackLocalePartOfSupportedLocales("en", generatedSupportedLocales)) {
+			log.warn("manifest.json: Generated supported locales ('" + generatedSupportedLocales.join("', '") + "') " +
+				"do not contain default fallback locale 'en'. Either provide a " +
+				"properties file for 'en' or configure another available fallbackLocale");
+		}
+
+		bundleConfig.supportedLocales = generatedSupportedLocales;
+	}));
+
+	resource.setString(JSON.stringify(manifest, null, options.prettyPrint ? "  " : undefined));
```

Comment:
I would prefer to use tabs for formatting, as that's the recommendation by UI5.
Also, I don't see the need for a non-pretty-printed file, but I'm okay to keep this option as long as it defaults to true and is documented properly.